server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -170,15 +170,26 @@ final private[worker] class StorageManager(conf: RssConf, workerSource: Abstract
   private var fileInfosDb: DB = null
   // ShuffleClient can fetch data from a restarted worker only
   // when the worker's fetching port is stable.
-  if (RssConf.workerGracefulShutdown(conf)) {
-    try {
-      val recoverFile = new File(RssConf.workerRecoverPath(conf), RECOVERY_FILE_INFOS_FILE_NAME)
-      this.fileInfosDb = LevelDBProvider.initLevelDB(recoverFile, CURRENT_VERSION)
-      reloadAndCleanFileInfos(this.fileInfosDb)
-    } catch {
-      case e: Exception =>
-        logError("Init level DB failed:", e)
-        this.fileInfosDb = null
+
+  {
+    if (RssConf.workerGracefulShutdown(conf)) {
+      try {
+        val recoverFile = new File(RssConf.workerRecoverPath(conf), RECOVERY_FILE_INFOS_FILE_NAME)
+        this.fileInfosDb = LevelDBProvider.initLevelDB(recoverFile, CURRENT_VERSION)
+        reloadAndCleanFileInfos(this.fileInfosDb)
+      } catch {
+        case e: Exception =>
+          logError("Init level DB failed:", e)
+          this.fileInfosDb = null
+      }
+    }
+    cleanupExpiredAppDirs(System.currentTimeMillis())
+    if(!checkIfWorkingDirCleaned) {
+      logWarning("Worker still has residual files in the working directory before registering with Master, " +
+        "please refer to the configuration document to increase rss.worker.checkIfFileCleanedRetryCount or " +
+        "rss.worker.checkIfFileCleanedTimeoutMs .")
+    } else {
+      logInfo("Successfully remove all files under working directory.")
     }
```

Comment:
Only clean when not enable graceful shutdown?
```
 if (RssConf.workerGracefulShutdown(conf)) {
      try {
        val recoverFile = new File(RssConf.workerRecoverPath(conf), RECOVERY_FILE_INFOS_FILE_NAME)
        this.fileInfosDb = LevelDBProvider.initLevelDB(recoverFile, CURRENT_VERSION)
        reloadAndCleanFileInfos(this.fileInfosDb)
      } catch {
        case e: Exception =>
          logError("Init level DB failed:", e)
          this.fileInfosDb = null
      }
    } else {
      cleanupExpiredAppDirs(System.currentTimeMillis())
      if(!checkIfWorkingDirCleaned) {
        logWarning("Worker still has residual files in the working directory before registering with Master, " +
          "please refer to the configuration document to increase rss.worker.checkIfFileCleanedRetryCount or " +
          "rss.worker.checkIfFileCleanedTimeoutMs .")
      } else {
        logInfo("Successfully remove all files under working directory.")
      }
    }
```

Reply:
We don't know if the last shutdown is a graceful shutdown, maybe this startup is the first time that graceful shutdown is enabled. Besides, even though last stop was graceful, there's a chance that data was not completed cleaned for reasons like timeout or others.